### PR TITLE
fix: force thorchain swapper to wait for outbound tx confirmations on-chain

### LINKED ISSUE #6086: Streaming swaps completion state errors
### summary

When attempting a streaming swap from an EVM to BTC after the initial EVM input transaction was confirmed, the UI showed as Trade Completed / BTC is now in your wallet and the progress bar was still spinning. Even after the BTC was confirmed the progress bar still didn't update.

This was not correct and was much too fast for the outgoing BTC tx to have already been executed. 

Will ask operations to recreate and record screen. 

### Recreation steps

1. swap BNB or other evm asset to BTC via streaming swap
2. when BNB tx confirms UI will update to a trade complete state incorreclty

### Acceptance Criteria

- Trade complete only shown when outgoing tx is confirmed
- progress bar is updated appropriately 

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Prevents thor trades from being displayed as complete until the outbound tx is confirmated on-chain.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #6086

Low risk, trade confirmation is already somewhat broken in that it completes early.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Check that thorchain trades only show as completed in the UI when they are in fact complete, not "pending outbound". Specifically thorchain trades that have an outbound tx, e.g `RUNE -> something`, or `BTC -> AVAX`. Not `something -> RUNE`.

# NOTE: dont use viewblock to test this, use our app tx history

App can show as complete, while viewblock shows `Pending: Outbound`. This is because they index the blockchain slower than us:

image
image
image

And sure enough after giving them a moment to catch up, it's confirmed there too
image

Note @shapeshift/operations - we can also use THORNode /tx/ endpoint to test that the outbound Tx has been signed 

## Screenshots (if applicable)

Outbound tx hash first seen, 0 confirmations. :
image

Trade continues to wait, with buy tx hash rendered and clickable. After some time, a confirmation is seen:
image

Trade is now shown as complete, in app and in viewblock for both txs:
image
image
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6143** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6143?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6143?utm_source=stack-comment).

File: src/lib/utils/index.ts [hunk 1]
```diff
@@ -205,7 +205,22 @@ export const isUrl = (x: string) => {
...
+    promise,
+    new Promise<FallbackType>(resolve =>
+      setTimeout(() => {
+        resolve(fallbackValue)
+      }, timeoutMs),
+    ),
+  ])
+}
+
+export const timeoutMonadic = <Left, Right>(
```
Comment (woodenfurniture) [inline src/lib/utils/index.ts:223]: ended up using axios in-built timeout but decided to keep these changes as they could be handy

File: src/lib/swapper/swappers/ThorchainSwapper/utils/parseThorBuyTxHash.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export const parseThorBuyTxHash = (
   // swaps into rune aren't double swaps so don't have a second tx (buy tx)
   if (!isDoubleSwap) return sellTxId
 
-  const isEvmCoinAsset = !THORCHAIN_EVM_CHAINS.some(
+  const isEvmCoinAsset = THORCHAIN_EVM_CHAINS.some(
```
Comment (woodenfurniture) [inline src/lib/swapper/swappers/ThorchainSwapper/utils/parseThorBuyTxHash.ts:16]: note bug fix, it was pre-pending a `0x` where it should not have (i.e UTXO tx hashes) due to inverted logic

File: src/lib/swapper/swapper.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import type {
 } from '@shapeshiftoss/swapper'
 import type { Asset, AssetsByIdPartial } from '@shapeshiftoss/types'
 import { bnOrZero } from 'lib/bignumber/bignumber'
-import { isSome, timeout } from 'lib/utils'
+import { isSome, timeoutMonadic } from 'lib/utils'
```
Comment (gomesalexandre) [inline src/lib/swapper/swapper.ts:10]: 👀

**Review by gomesalexandre — COMMENTED** [review]
First pass, this looks good to me at a conceptual level pending runtime testing

**Review by gomesalexandre — APPROVED** [review]
Runtime testing:

### Swap to RUNE

image

- Confirmed this is still happy and completes pretty much instantly after inbound is seen since inner swaps to RUNE are effectively instant and do not trigger L1 outbounds

### Swaps to EVM chain

- `isEvmCoinAsset` now properly returns `true` for EVM chains 🎉 

image

- outbound confirmations are properly detected

image

- trade properly goes to confirmed state after on-chain confirmation

image

### Swaps to UTXOs

- Trade is properly detected as complete after on-chain confirmation

image

### Swap to ATOM

image

Note, I wasn't able to get to a confirmed state after outbound signed and on-chain confirmation, but logic is the same here, the issue here is there's a transient issue with unchained:

image

**Pull request merged**